core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.spec.ts

```diff
@@ -0,0 +1,485 @@
+                        param_changes: [],
+                        labelled_as_correct: true
+                      },
+                      hints: [{
+                        hint_content: {
+                          html: 'Hint 1',
+                          content_id: 'content_3'
+                        }
+                      }],
+                      solution: {
+                        correct_answer: 'This is the correct answer',
+                        answer_is_exclusive: false,
+                        explanation: {
+                          html: 'Solution explanation',
+                          content_id: 'content_4'
+                        }
+                      },
+                      id: 'TextInput'
+                    },
+                    param_changes: [],
+                    recorded_voiceovers: {
+                      voiceovers_mapping: {}
+                    },
+                    written_translations: {
+                      translations_mapping: {}
+                    },
+                  },
+                }
+              },
+              status: 'review'
+            },
+            details: {
+              skill_description: 'Skill description'
+            }
+          }
+        }));
+
+      $scope = $rootScope.$new();
+      ctrl = $componentController('contributionsAndReview', {
+        $scope: $scope,
+        ContextService: contextService,
+        MisconceptionObjectFactory: misconceptionObjectFactory
+      });
+      ctrl.$onInit();
+      $scope.$apply();
+      $scope.$apply();
+    }));
+
+    it('should initialize $scope properties after controller is' +
+      ' initialized', function() {
+      expect(ctrl.activeReviewTab).toBe('add_question');
+      expect(ctrl.activeContributionTab).toBe('');
+      expect(ctrl.userIsLoggedIn).toBe(true);
+      expect(ctrl.userDetailsLoading).toBe(false);
+      expect(ctrl.reviewTabs.length).toEqual(2);
+      expect(Object.keys(ctrl.contributions)).toContain('suggestion_1');
+      expect(ctrl.contributionSummaries).toEqual([{
+        id: 'suggestion_1',
+        heading: 'Question 1',
+        subheading: 'Skill description',
+        labelText: 'Awaiting review',
+        labelColor: '#eeeeee',
+        actionButtonTitle: 'Review'
+      }]);
+      expect(ctrl.contributionsDataLoading).toBe(false);
+    });
+
+    it('should open show translation suggestion modal when clicking on' +
+      ' suggestion', function() {
+      ctrl.switchToContributionsTab('translate_content');
+      $scope.$apply();
+
+      spyOn($uibModal, 'open').and.callThrough();
+      ctrl.onClickViewSuggestion('suggestion_1');
+
+      expect($uibModal.open).toHaveBeenCalled();
+    });
+
+    it('should resolve suggestion when closing show suggestion modal',
+      function() {
+        ctrl.switchToContributionsTab('translate_content');
+        $scope.$apply();
+
+        spyOn(contributionAndReviewService, 'resolveSuggestiontoExploration');
```

Comment:
```suggestion
        spyOn(contributionAndReviewService, 'resolveSuggestionToExploration');
```
Can you make this change in the main code?

Reply:
Done.